src/main/java/org/opensearch/flowframework/workflow/CreateIngestPipelineStep.java

```diff
@@ -77,122 +68,85 @@ public PlainActionFuture<WorkflowData> execute(
-                        break;
-                    case INPUT_FIELD_NAME:
-                        inputFieldName = (String) content.get(INPUT_FIELD_NAME);
-                        break;
-                    case OUTPUT_FIELD_NAME:
-                        outputFieldName = (String) content.get(OUTPUT_FIELD_NAME);
-                        break;
-                    default:
-                        break;
-                }
-            }
-
-            // Determine if fields have been populated, else iterate over remaining workflow data
-            if (Stream.of(pipelineId, description, modelId, type, inputFieldName, outputFieldName).allMatch(x -> x != null)) {
-                try {
-                    configuration = BytesReference.bytes(
-                        buildIngestPipelineRequestContent(description, modelId, type, inputFieldName, outputFieldName)
-                    );
-                } catch (IOException e) {
-                    String errorMessage = "Failed to create ingest pipeline configuration for " + currentNodeId;
-                    logger.error(errorMessage, e);
-                    createIngestPipelineFuture.onFailure(new FlowFrameworkException(errorMessage, ExceptionsHelper.status(e)));
-                }
-                break;
-            }
-        }
+        Set<String> requiredKeys = Set.of(PIPELINE_ID, CONFIGURATIONS);
+
+        // currently, we are supporting an optional param of model ID into the various processors
+        Set<String> optionalKeys = Set.of(MODEL_ID);
 
-        if (configuration == null) {
-            // Required workflow data not found
-            createIngestPipelineFuture.onFailure(
-                new FlowFrameworkException(
-                    "Failed to create ingest pipeline for " + currentNodeId + ", required inputs not found",
-                    RestStatus.BAD_REQUEST
-                )
+        try {
+            Map<String, Object> inputs = ParseUtils.getInputsFromPreviousSteps(
+                requiredKeys,
+                optionalKeys,
+                currentNodeInputs,
+                outputs,
+                previousNodeInputs,
+                params
             );
-        } else {
-            // Create PutPipelineRequest and execute
-            PutPipelineRequest putPipelineRequest = new PutPipelineRequest(pipelineId, configuration, XContentType.JSON);
-            clusterAdminClient.putPipeline(putPipelineRequest, ActionListener.wrap(response -> {
-                logger.info("Created ingest pipeline : " + putPipelineRequest.getId());
 
-                try {
+            String pipelineId = (String) inputs.get(PIPELINE_ID);
+            String configurations = (String) inputs.get(CONFIGURATIONS);
+
+            byte[] byteArr = configurations.getBytes(StandardCharsets.UTF_8);
+            BytesReference configurationsBytes = new BytesArray(byteArr);
+
+            ActionListener<AcknowledgedResponse> actionListener = new ActionListener<>() {
```

Comment:
You could just define this inline on line 146, where it'd be a bit more logically readable.

Reply:
moved